jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/HadoopIntegrationTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper.target_systems;
+
+import static io.opentelemetry.contrib.jmxscraper.target_systems.MetricAssertions.assertSumWithAttributes;
+import static org.assertj.core.api.Assertions.entry;
+
+import io.opentelemetry.contrib.jmxscraper.JmxScraperContainer;
+import java.time.Duration;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+public class HadoopIntegrationTest extends TargetSystemIntegrationTest {
+
+  @Override
+  protected GenericContainer<?> createTargetContainer(int jmxPort) {
+    return new GenericContainer<>("bmedora/hadoop:2.9-base")
+        .withCopyFileToContainer(
+            MountableFile.forClasspathResource("hadoop-env.sh", 0400),
+            "/hadoop/etc/hadoop/hadoop-env.sh")
+        .waitingFor(Wait.forListeningPort().withStartupTimeout(Duration.ofMinutes(200)))
+        .withExposedPorts(jmxPort)
+        .withCreateContainerCmdModifier(cmd -> cmd.withHostName("test-host"))
+        .waitingFor(Wait.forListeningPort());
```

Comment:
In other integration tests we started also exposing and waiting for some service point of target application. It makes the tests more stable. Please identify some port that is opened by Hadoop (other than JMX ports), and ezpose it and wait for it, like for example in ActiveMQ or Cassandra.